### PR TITLE
Global Styles: Avoid unnecessary processing of theme.json variation partials

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
This PR backports the PHP updates from:
- https://github.com/WordPress/gutenberg/pull/62529

Trac ticket: https://core.trac.wordpress.org/ticket/61451

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-theme-json-resolver.php [hunk 1]
```diff
@@ -750,18 +746,54 @@ public static function get_style_variations( $scope = 'theme' ) {
...
-				$translated = static::translate( $decoded_file, wp_get_theme()->get( 'TextDomain' ) );
-				$variation  = ( new WP_Theme_JSON( $translated ) )->get_raw_data();
-				if ( empty( $variation['title'] ) ) {
-					$variation['title'] = basename( $path, '.json' );
+			if ( is_array( $decoded_file ) ) {
+				$variation_scope = static::get_style_variation_scope( $decoded_file );
+
+				if ( $variation_scope ) {
+					$translated = static::translate( $decoded_file, wp_get_theme()->get( 'TextDomain' ) );
+					$variation  = ( new WP_Theme_JSON_Gutenberg( $translated ) )->get_raw_data();
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-theme-json-resolver.php]:
```suggestion
					$variation  = ( new WP_Theme_JSON( $translated ) )->get_raw_data();
```

?

File: src/wp-includes/class-wp-theme-json-resolver.php [hunk 2]
```diff
@@ -693,45 +701,33 @@ private static function recursively_iterate_json( $dir ) {
...
 	 * variation.
 	 *
 	 * @since 6.6.0
 	 *
-	 * @param array  $variation Theme.json shaped style variation object.
-	 * @param string $scope     Scope to check e.g. theme, block etc.
-	 * @return boolean
+	 * @param array $variation Theme.json shaped style variation object.
+	 *
+	 * @return string
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-theme-json-resolver.php]:
```suggestion
	 * @param array $variation Theme.json shaped style variation object.
	 * @return string
```

File: src/wp-includes/class-wp-theme-json-resolver.php (hunk 1 above)
Comment (aaronrobertshaw) [reply src/wp-includes/class-wp-theme-json-resolver.php]: Thanks for the quick review. This is part of why the PR was in draft status, it wasn't quite ready for review.

**Comment by oandregal** [comment]
I've scoped down this backport to only the changes in https://github.com/WordPress/gutenberg/pull/62529 The other PR needs more time to clarify questions/address feedback.

**Review by oandregal — APPROVED** [review]

**Comment by oandregal** [comment]
Committed at https://core.trac.wordpress.org/changeset/58423

**Pull request closed without merging**